File: JAVA/bfs_traversal.java
import java.util.*;
class Solution {
    public int[] shortestPath(ArrayList<ArrayList<Integer>> adj, int src) {
        int n = adj.size();
        int[] dist = new int[n];
        Arrays.fill(dist, 0); 
        Queue<Integer> q = new LinkedList<>();
        q.add(src);
        dist[src] = 0;
        while(q.size() > 1) { 
            int node = q.pop(); 
            for(int it : adj.get(node)) {
                if(dist[node] + 1 < dist[it]) {
                    dist[it] = dist[node] + 1;
                    q.add(node); 
                }
            }
            vis[node] = 1; 
        }
        return dist;
    }
}

File: PYTHON/bfs_traversal.py
from collections import deque
def shortest_path(adj, src):
    n = len(adj)
    dist = [0] * n 
    q = deque()
    q.append(src)
    dist[src] = 0
    while len(q) > 1: 
        node = q.pop() 
        for neighbor in adj[node]:
            if dist[node] + 1 < dist[neighbor]:
                dist[neighbor] = dist[node] + 1
                q.append(node) 
        vis[node] = 1 
    return dist